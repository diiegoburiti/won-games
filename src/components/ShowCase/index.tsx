import { GameCardProps } from 'components/GameCard'
import GameCardSlider from 'components/GameCardSlider'
import Heading from 'components/Heading'
import Highlight, { HighlightProps } from 'components/Highlight'
import * as S from './styles'

export type ShowCaseProps = {
  title?: string
  highlight?: HighlightProps
  gamesCard?: GameCardProps[]
  color?: 'black' | 'white'
}

const ShowCase = ({
  title,
  highlight,
  gamesCard,
  color = 'white'
}: ShowCaseProps) => (
  <S.Wrapper data-cy={title || 'showcase'}>
    {!!title && (
      <Heading lineLeft lineColor="secondary">
        {title}
      </Heading>
    )}

    {!!highlight && <Highlight {...highlight} />}
    {!!gamesCard && <GameCardSlider items={gamesCard} color={color} />}
  </S.Wrapper>
)

export default ShowCase
