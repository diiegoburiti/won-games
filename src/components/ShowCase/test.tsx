import 'match-media-mock'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import highlightMock from 'components/Highlight/mock'
import gamesMock from 'components/GameCardSlider/mock'

import ShowCase from '.'

const props = {
  title: 'Most Popular',
  highlight: highlightMock,
  gamesCard: gamesMock.slice(0, 1)
}

describe('<ShowCase />', () => {
  it('should render ShowCase', () => {
    const { container } = renderWithTheme(<ShowCase {...props} />)

    expect(
      screen.getByRole('heading', { name: /Most Popular/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: gamesMock[1].title })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: highlightMock.title })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render ShowCase without title', () => {
    renderWithTheme(
      <ShowCase gamesCard={props.gamesCard} highlight={props.highlight} />
    )

    screen.getByRole('heading', { name: gamesMock[1].title })

    screen.getByRole('heading', { name: highlightMock.title })

    expect(
      screen.queryByRole('heading', { name: /most Popular/i })
    ).not.toBeInTheDocument()
  })

  it('should render ShowCase without highlight', () => {
    renderWithTheme(
      <ShowCase gamesCard={props.gamesCard} title={props.title} />
    )

    expect(
      screen.getByRole('heading', { name: gamesMock[1].title })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /most Popular/i })
    ).toBeInTheDocument()

    expect(
      screen.queryByRole('heading', { name: highlightMock.title })
    ).not.toBeInTheDocument()
  })
})
