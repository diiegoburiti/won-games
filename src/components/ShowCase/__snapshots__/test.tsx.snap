// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowCase /> should render ShowCase 1`] = `
.c25 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c10:focus {
  outline: 1px dashed;
}

.c10:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c24 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 3rem;
  font-size: 1.2rem;
}

.c24:focus {
  outline: 1px dashed;
}

.c24:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c24 svg {
  width: 1.5rem;
}

.c24 svg + span {
  margin-left: 0.8rem;
}

.c14 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: #FAFAFA;
  -webkit-transition: -webkit-transform 0.5s;
  -webkit-transition: transform 0.5s;
  transition: transform 0.5s;
}

.c14 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c14:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c15 {
  height: 14rem;
  width: 100%;
  background: #f6f7f8;
  background-image: linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100% );
  background-size: 80rem 14rem;
  -webkit-animation: placeholderShimmer 1s linear infinite forwards;
  animation: placeholderShimmer 1s linear infinite forwards;
}

.c15 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  height: 100%;
  margin: 1.6rem;
}

.c17 {
  max-width: calc(100% - 2.5rem);
}

.c18 {
  font-size: 1.6rem;
  line-height: 1.6rem;
  font-weight: 600;
  color: #030517;
}

.c19 {
  font-size: 1.4rem;
  font-weight: 600;
  color: #8F8F8F;
}

.c20 {
  color: #F231A5;
  position: absolute;
  right: -1rem;
  top: -0.5rem;
  cursor: pointer;
}

.c20 svg {
  width: 2.5rem;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 0.8rem;
}

.c22 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #8F8F8F;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  margin-right: 1.6rem;
}

.c23 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 3rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FAFAFA;
  padding: 0 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  margin-right: calc(0.8rem / 2);
}

.c13 .slick-slider {
  position: relative;
  display: block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.c13 .slick-list {
  position: relative;
  display: block;
  overflow: hidden;
  margin: 0;
  padding: 0;
}

.c13 .slick-list:focus {
  outline: none;
}

.c13 .slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.c13 .slick-slider .slick-track,
.c13 .slick-slider .slick-list {
  -webkit-transform: translate3d(0,0,0);
  -moz-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  -o-transform: translate3d(0,0,0);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c13 .slick-track {
  position: relative;
  top: 0;
  left: 0;
  display: block;
}

.c13 .slick-track:before,
.c13 .slick-track:after {
  display: table;
  content: '';
}

.c13 .slick-track:after {
  clear: both;
}

.c13 .slick-loading .slick-track {
  visibility: hidden;
}

.c13 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

.c13 [dir='rtl'] .slick-slide {
  float: right;
}

.c13 .slick-slide img {
  display: block;
}

.c13 .slick-slide.slick-loading img {
  display: none;
}

.c13 .slick-slide.dragging img {
  pointer-events: none;
}

.c13 .slick-initialized .slick-slide {
  display: block;
}

.c13 .slick-loading .slick-slide {
  visibility: hidden;
}

.c13 .slick-vertical .slick-slide {
  display: block;
  height: auto;
  border: 1px solid transparent;
}

.c13 .slick-arrow.slick-hidden {
  display: none;
}

.c12 .slick-track,
.c12 .slick-list {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 .slick-slide > div {
  margin: 0 0.8rem;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  height: 100%;
}

.c12 .slick-list {
  margin: 0 -0.8rem;
}

.c12 .slick-prev,
.c12 .slick-next {
  display: block;
  color: #FAFAFA;
  cursor: pointer;
  position: absolute;
  top: 50%;
  width: 2.5rem;
  height: 2.5rem;
  padding: 0;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

.c12 .slick-prev {
  left: -5.6rem;
}

.c12 .slick-next {
  right: -5.6rem;
}

.c12 .slick-prev.slick-disabled,
.c12 .slick-next.slick-disabled {
  visibility: hidden;
}

.c3 {
  color: #FAFAFA;
  padding-left: 0.8rem;
  border-left: 0.7rem solid #3CD3C1;
  font-size: 2.0rem;
}

.c5 {
  position: relative;
  background-image: url(/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
  height: 23rem;
  display: grid;
  grid-template-areas: 'floatimage content';
  grid-template-columns: 1.3fr 2fr;
}

.c5::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.6);
}

.c5 .c6 {
  text-align: right;
}

.c7 {
  grid-area: content;
  z-index: 10;
  padding: 1.6rem;
}

.c8 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c9 {
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
  margin-bottom: 3.2rem;
}

.c0 {
  width: 100%;
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c1 {
  margin-bottom: calc(4.0rem * 2);
}

.c1 .c2,
.c1 .c4,
.c1 .c11 {
  margin-bottom: 3.2rem;
}

@media (max-width:1440px) {
  .c12 {
    overflow-x: hidden;
  }
}

@media (min-width:1170px) {
  .c12 .slick-slide > div {
    margin: 0 1.6rem;
  }

  .c12 .slick-list {
    margin: 0 -1.6rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c5 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c7 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c8 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c9 {
    font-size: 1.8rem;
  }
}

@media (max-width:768px) {
  .c1 .c4 {
    margin-right: calc(-3.2rem / 2);
    margin-left: calc(-3.2rem / 2);
  }
}

@media (max-width:1440px) {
  .c1 .c11 {
    margin-right: calc(-3.2rem / 2);
  }
}

<section
  class="c0 c1"
  data-cy="Most Popular"
>
  <h2
    class="c2 c3"
    color="white"
  >
    Most Popular
  </h2>
  <section
    class="c4 c5"
    data-cy="highlight"
  >
    <div
      class="c6 c7"
    >
      <h2
        class="c8"
      >
        Red Dead is back
      </h2>
      <h3
        class="c9"
      >
        Come see jhon\`s adventures
      </h3>
      <a
        class="c10"
        href="/rd2"
      >
        <span>
          Buy now
        </span>
      </a>
    </div>
  </section>
  <section
    class="c11 c12"
    color="white"
  >
    <section
      class="c13"
    >
      <div
        class="slick-slider slick-initialized"
      >
        
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <article
                  class="c14"
                  data-cy="game-card"
                >
                  <a
                    class="c15"
                    href="/game/population-zero"
                  >
                    <img
                      alt="Population Zero"
                      src="https://source.unsplash.com/user/willianjusten/300x140"
                    />
                  </a>
                  <div
                    class="c16"
                  >
                    <a
                      class="c17"
                      href="/game/population-zero"
                    >
                      <h3
                        class="c18"
                      >
                        Population Zero
                      </h3>
                      <h4
                        class="c19"
                      >
                        Rockstar Games
                      </h4>
                    </a>
                    <div
                      class="c20"
                    />
                    <div
                      class="c21"
                    >
                      <div
                        class="c22"
                      >
                        $235.00
                      </div>
                      <div
                        class="c23"
                      >
                        $215.00
                      </div>
                      <button
                        aria-label="Add To Cart"
                        class="c24"
                      >
                        <svg
                          aria-hidden="true"
                          class="c25"
                          fill="currentColor"
                          focusable="false"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 0h24v24H0V0z"
                            fill="none"
                          />
                          <path
                            d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </div>
        
        
      </div>
    </section>
  </section>
</section>
`
